tests/node_suspension.py

```diff
@@ -0,0 +1,157 @@
+            )
+            tm1.start()
+            tm2 = Timer(
+                t2, timeout, args=[backups[1], True, args.election_timeout / 1000]
+            )
+            tm2.start()
+
+            with first_node.node_client() as mc:
+                check_commit = infra.ccf.Checker(mc, notifications_queue)
+                check = infra.ccf.Checker()
+
+                LOG.info("Write messages to nodes using round robin")
+                with first_node.user_client(format="json") as c0:
+                    with backups[0].user_client(format="json") as c1:
+                        with backups[1].user_client(format="json") as c2:
+                            node_id = 0
+                            for id in range(1, TOTAL_REQUESTS):
+                                node_id += 1
+                                node_id %= 3
+                                if node_id == 0:
+                                    c = c0
+                                elif node_id == 1:
+                                    c = c1
+                                else:
+                                    c = c2
+                                try:
+                                    resp = c.rpc(
+                                        "LOG_record", {"id": id, "msg": long_msg}
+                                    )
+                                except Exception:
+                                    LOG.info("Trying to access a suspended node")
+                                try:
+                                    cur_primary, cur_term = network.find_primary()
+                                    term_info[cur_term] = cur_primary.node_id
+                                except Exception:
+                                    LOG.info("Trying to access a suspended node")
+                                id += 1
+
+                            # wait for the last request to commit
+                            final_msg = "Hello world!"
+                            check_commit(
+                                c.rpc("LOG_record", {"id": 1000, "msg": final_msg}),
+                                result=True,
+                            )
+                            check(
+                                c.rpc("LOG_get", {"id": 1000}),
+                                result={"msg": final_msg},
+                            )
+
+                            # check that a new node can catch up after all the requests
+                            new_node = network.create_and_trust_node(
+                                lib_name=args.package,
+                                host="localhost",
+                                args=args,
+                                should_wait=False,
+                            )
+                            assert new_node
+
+                            with new_node.user_client(format="json") as c:
+                                while True:
+                                    rep = c.do("LOG_get", {"id": 1000})
```

Comment:
passing expected_result={"msg": final_msg} to do() is slightly more precise than the next two lines.